File: Data Structures & Algorithms/two-integer-sum/submission-2.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> seen = new HashMap<>();
        int [] ans = new int[2];
        for(int i=0;i<nums.length;i++){
            int num = nums[i];
            int diff = target - num;
            if(seen.containsKey(diff)){
                ans[0]=seen.get(diff);
                ans[1]=i;
                return ans;
            }
            seen.put(num,i);

        }
        return new int[]{-1,-1};
        
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        ans=[]
        l=0
        r=len(numbers)-1
        sum = numbers[l] + numbers[r]
        while l<r:
            if numbers[l] + numbers[r]<target:
                l+=1
            elif numbers[l] + numbers[r]>target:
                r-=1
            else:
                ans.append(l+1)
                ans.append(r+1)
                l+=1
                r-=1
        return ans
